api/src/pcapi/core/offerers/api.py

```diff
@@ -2321,5 +2323,54 @@ def match_venue_with_new_entries(
 
 
 def update_offerer_address_label(offerer_address_id: int, new_label: str) -> None:
-    models.OffererAddress.query.filter_by(id=offerer_address_id).update({"label": new_label})
-    db.session.flush()
+    try:
+        models.OffererAddress.query.filter_by(id=offerer_address_id).update({"label": new_label})
+        db.session.flush()
+    except sa.exc.IntegrityError:
+        raise exceptions.OffererAddressLabelAlreadyUsed()
+
+
+def get_or_create_address(address_info: AddressInfo) -> geography_models.Address:
+    try:
+        address = geography_models.Address(
+            banId=address_info.id,
+            inseeCode=address_info.citycode,
+            street=address_info.street,
+            postalCode=address_info.postcode,
+            city=address_info.city,
+            latitude=address_info.latitude,
+            longitude=address_info.longitude,
+        )
+        db.session.add(address)
+        db.session.flush()
+    except sa.exc.IntegrityError:
+        db.session.rollback()
+        address = geography_models.Address.query.filter(
+            geography_models.Address.street == address_info.street,
+            geography_models.Address.inseeCode == address_info.citycode,
+        ).one()
+        if (address.latitude, address.longitude) != (address_info.latitude, address.longitude):
```

Comment:
Je vais être casse-pieds : a-t-on un risque de différence due à l'arrondi ?

Mais aussi : 
```suggestion
        if (address.latitude, address.longitude) != (address_info.latitude, address_info.longitude):
```

Reply:
oui effectivement, on arrondi après 5 digits, j'ai corrigé, merci !